front/lib/api/assistant/actions/visualization.ts

```diff
@@ -200,6 +201,46 @@ export class VisualizationConfigurationServerRunner extends BaseActionConfigurat
       return;
     }
 
+    const renderedConversation = modelConversationRes.value.modelConversation;
+    const lastMessage =
+      renderedConversation.messages[renderedConversation.messages.length - 1];
+
+    // If the last message is from the assistant, it means some content might have been generated along with the visualization tool call.
+    // In this case, we want to preserver the content (as it may be relevant to the model, as a "chain of thought"), so we
+    // update the last message to include an "enable_visualization_mode" function call and a dummy "visualization mode enabled" function
+    // result.
+    // This is necessary, as some model providers like Anthropic do not support multiple agent messages in a row.
+    if (lastMessage.role === "assistant") {
+      renderedConversation.messages.pop();
+
+      // Only bother with inserting tool_call/tool_result if there is content to preserve.
+      if (lastMessage.content) {
```

Comment:
yeah it can technically happen although unlikely

> In such a case we simply drop this last agent message?

Yep it's exactly what I do here: we pop it above, and we only replace with tool_call/result if the content is truthy